RedisChecker/redisChecker.go

```diff
@@ -35,18 +42,44 @@ func (m *RedisConnection) Validate() (bool, error) {
 func (m *RedisConnection) Check() (bool, bool, error) {
 	ctx := context.Background()
 
-	rdb := redis.NewClient(&redis.Options{
-		Addr:     m.BuildConnectionString(),
-		Password: m.Password, // no password set
-		DB:       m.Database, // use default DB
-	})
+	if !m.IsClustered {
+		rdb := redis.NewClient(&redis.Options{
+			Addr:     m.BuildConnectionString(),
+			Password: m.Password, // no password set
+			DB:       m.Database, // use default DB
+		})
 
-	_, err := rdb.Ping(ctx).Result()
-	if err != nil {
-		return false, false, nil
-	}
+		_, err := rdb.Ping(ctx).Result()
+		if err != nil {
+			return false, false, nil
+		}
+
+		_ = rdb.Close()
 
-	_ = rdb.Close()
+		return true, true, nil
+	} else {
+		rdb := redis.NewClusterClient(&redis.ClusterOptions{
+			Addrs:              []string{m.BuildConnectionString()},
+			Password:           m.Password,
+		})
+		defer rdb.Close()
 
-	return true, true, nil
+		_, err := rdb.Ping(ctx).Result()
+		if err != nil {
+			return false, false, nil
+		}
+
+		result, err := rdb.ClusterInfo(ctx).Result()
+		if err != nil {
+			return false, false, nil
+		}
+
+		if result != "" {
```

Comment:
You can merge the two if in a one if statement, right?

Reply:
I think merging those two if statements decrease code verbosity and readability. because the return statements are different and merging two ifs causes indentation and I think it's a bad practice.